src/twisted/internet/interfaces.py

```diff
@@ -2707,3 +2710,130 @@ class _ISupportsExitSignalCapturing(Interface):
         application, or None if no signal was delivered.
         """
     )
+
+
+DEFAULT_CHUNK_SIZE = 4096
+DEFAULT_BUFFER_MAX = 2 ** 22  # 4 megabytes
+
+
+class IAsyncReader(Interface):
+    """
+    a file or file-like object opened for reading asynchronously
+    """
+
+    def close() -> "Deferred":
```

Comment:
I think this should return an Awaitable or a Coroutine

Reply:
```suggestion
    def aclose() -> Awaitable[object]:
```

then it can be used in ```async with aclosing(reader):```

Reply:
IAsyncReader should also be a subtype of contextlib.AsyncContextManager[T]